### ISSUE TITLE
Celsius air conditioners will not set temperature

### ISSUE THREAD

**Issue by amaisano** [bm1549/frigidaire#19]
https://github.com/bm1549/frigidaire/blob/3b8e3541f69eb1ad878d18dc5a4b070702ad8534/frigidaire/__init__.py#L211-L214

Why is this hard coded to Fahrenheit? My AC units are all set (individual settings in the 2.0 app, and in the profile settings of the 3.0 2023 app) to show as Celsius.

(Yes, there are many people in the US that prefer Celsius, instead of Fahrenheit)

Home assistant is also set to Celsius.

When I try to request 24° C, I get this error:

> Temperature must be between 60 and 90 degrees, inclusive

Which makes no sense. I think we need a check for units, but I'm having a hard time determining if we can get that info reliably from the API. If not, maybe a user variable setting?

**Comment by amaisano** [comment]
Bump

**Comment by rothn** [comment]
https://github.com/bm1549/frigidaire/pull/22 would make this possible to specify manually as a user:

```
fr_api.execute_action(my_appliance, [
        frigidaire.Component(Setting.TARGET_TEMPERATURE_C, temperature)
])
```

A cleaner way to address this upstream might be to make `set_temperature()` take an `Optional[Unit]` that defaults to fahrenheit (to preserve backward compatibility, e.g. for the homeassistant integration).

**Comment by amaisano** [comment]
bump

**Comment by amaisano** [comment]
why is everything hardcoded in `F` ???

**Comment by rothn** [comment]
The maintainers are most likely from North America-- I know I am-- and too busy with other things in our personal lives to add functionality that doesn't benefit us (again, I know I am). It should be a relatively simple change, as detailed by my comments above.

**Comment by bm1549** [comment]
Closed via #34
